### DISCUSSION TITLE
Sharing improvement

### DISCUSSION THREAD

**Discussion by Hutch79** [mealie-recipes/mealie#2697] [Feature Request]
Rewrite since I missed a key feature ^^'

### Please provide a concise description of the problem that would be addressed by this feature.
Currently, there are two buttons to share a recipe.
A Public link button and a Share button, which can be accessed by clicking the three dots on a recipe.
![grafik](https://github.com/mealie-recipes/mealie/assets/42042811/a722a276-c79f-4016-b3d2-661d34e9b133)

This can lead to confusion, especially if your group is Private.
A non-tech person sees the public link button and probably expects the link to be accessible by everyone since it is public.

### Please provide a concise description of the feature that would resolve your issue.
My suggestion to solve this problem is to combine the "Share" and "Public Link" into one share feature.

The current share feature already contains expiration dates, which is great!
This could be expanded with a "share internal" and a "share public" checkbox.
Also, a grate option would be to disable the Expiration date, or set it to the year 3023 internally.
![grafik](https://github.com/mealie-recipes/mealie/assets/42042811/0485296a-baf5-4dbf-b5dd-c92d9e2b972e)

The internal and public sharing concept may need new group settings for sharing, like disable public sharing. If such an option is integrated, I would also suggest to not show the option in the share Recipe popup.

---
I will leave the following as it is the original discussion I've started, although its irrelevant since i missed that there is a group setting for public sharing already.

> ### First Check
> 
> - [X] I used the GitHub search to find a similar requests and didn't find it.
> - [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered
> 
> ### Please provide a concise description of the problem that would be addressed by this feature.
> 
> Currently, if you share a public link, you revel the URL where someone could access all your recipes.
> Here is an example link from the password protected Demo https://demo.mealie.io/explore/recipes/home/
> Although I'm not logged in, I can access all recipes without restrictions.
> 
> ### Please provide a concise description of the feature that would resolve your issue.
> 
> There already is a system to share recipes without this path, the normal share feature where you can set a time limit.
> 
> My suggestion would be to add an infinite option (if not possible, it expires at 3023 or something like that) to the sharing menu.
> This would allow sharing, without the potential of others finding all your recipes.
> 
> This would also combine two different sharing features into one single feature, which makes the dropdown smaller and thus a bit simpler to use.
> 
> ### Please consider and list out some caveats or tradeoffs made in your design decision
> 
> Alternatively the `/explore/recipes/home` path could also be password protected.
> This solution may even be the better option, since you can recognize a recipe because of the link itself.
> But I would still suggest combining the two sharing features (share and public link).
> 
> @michael-genson showed me the option to set this site to private.
> ~~If the publication of `/explore/recipes/home` is intended, an option to turn this off would be great.~~
> 
> ### Additional Information
> 
> - [ ] If this is accepted I'm willing to submit a PR to provide this feature
> - [ ] If this is accepted I'm willing to help maintain this feature
> - [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
> If the publication of /explore/recipes/home is intended, an option to turn this off would be great

Well I was _going_ to show you which option to toggle to disable the public explore link, but it looks like I used the wrong option. Oops. I'll fix this in https://github.com/mealie-recipes/mealie/pull/2632

In the meantime, you can toggle off this option to disable the explore link:
![image](https://github.com/mealie-recipes/mealie/assets/71845777/fcc1f38e-0f5e-4b5d-9c5f-8b7e6ac29128)

It _should_ be this option combined with the above option, but it doesn't work (which is a bug):
![image](https://github.com/mealie-recipes/mealie/assets/71845777/78d2b325-d981-49fa-902b-ac9955fb9fa4)

**Reply by Hutch79** [reply]
This changes a lot!
I think it's time for a rewrite ^^

Thanks very much!

**Comment by Rafficer** [comment]
I think the request to have unlimited sharing would still be great. My mealie instance is publicly available and I use it to share my recipes with others if I'm asked to. I don't know when they'd want to use the recipe, so an unlimited link would be great!

**Reply by Kuchenpirat** [reply]
If your instance is public you can just share the link. If you don't change the recipe title the link should not change.

**Reply by Rafficer** [reply]
That's true, but you can't create non-expiring links. Eventually they will expire and people I shared them with can't access it anymore.

The ability to not have them expire is what I meant :)

Sorry, unlimited was a really bad choice of words.

**Reply by Kuchenpirat** [reply]
If you just copy the link from the in the Adress Bar of your browser this should not expire.

**Reply by Rafficer** [reply]
Huh? What is the expiration setting for then? Got to test that tomorrow, if that really doesn't expire.

**Reply by michael-genson** [reply]
Share links are only for granting access to private instances. If your instance is public, your group is public, your household is public, and the recipe you want to share is public, anyone can access the recipe with the regular URL.

Moreover, anyone can access your homepage and browse your recipes if your group is public. Private recipes (or recipes that belong to private households) will be hidden from them.

More info in the docs: https://docs.mealie.io/documentation/getting-started/usage/permissions-and-public-access/

**Reply by Rafficer** [reply]
I think we have another misunderstanding here. What I mean with "publicly available" is that it's reachable over the internet. But when accessing it, all you get is the login form, so you can't actually just browse the recipes. Which is also what I want.

So when I copy the URL from the address bar, other people won't have access, they just get a login form.

But when I use the share button to create a link like this it will work:

![image](https://github.com/user-attachments/assets/1cbdbace-0197-4071-810b-7b9ee82b25b0)

That method results in a link like https://mealie.example.com/g/home/shared/r/d1b7167a-a9f9-4a8c-baf9-4251fe8cd73f

But there I can't set it to not expire. Or does it just say it will expire but then doesn't actually expire? I'm a bit confused now as to what that's actually for.

**Reply by michael-genson** [reply]
Were you able to check out the documentation I linked above? That should directly answer your question about making your recipes available to visitors without requiring a login

**Reply by Rafficer** [reply]
I did, and like I said, it works. I can share recipes using the share button with no issue, my problem relies in those links to expire:

If I send someone a link using the default values (expiration in 30 days) and that person accesses the link not immediately but after 2 months, the link won't work anymore, correct?

My request is to have an option to set the expiration of a private link to unlimited, so it never expires. Or is that somehow possible that I am not aware of?

**Reply by michael-genson** [reply]
Sorry, I think I misunderstood this comment:
> But when accessing it, all you get is the login form, so you can't actually just browse the recipes. Which is also what I want.

I read that as "I want users to be able to browse the recipes without logging in" but now I see you meant the opposite (I think).

In that case, yes, I think adding an option for share links not to expire makes sense
